### PR TITLE
ZEPPELIN-3106. User impersonation in SPARK is not working

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
This PR is for trying to fix the impersonation of spark interpreter in master branch.  For spark impersonation, we don't need to ssh, just adding `--proxy-user` is enough. 

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3106

### How should this be tested?
* Manually verified. 

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@prabhjyotsingh Could you help review it ? Thanks

**Comment by prabhjyotsingh** [comment]
Will try this for both spark and shell interpreter, and post my finding later today.

File: bin/interpreter.sh [hunk 1]
```diff
@@ -206,15 +210,12 @@ if [[ ! -z "$ZEPPELIN_IMPERSONATE_USER" ]]; then
 fi
 
 if [[ -n "${SPARK_SUBMIT}" ]]; then
-    if [[ -n "$ZEPPELIN_IMPERSONATE_USER" ]] && [[ "$ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER" != "false" ]];  then
```
Comment (prabhjyotsingh) [inline bin/interpreter.sh]: By removing this `ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER` you disable the feature that was introduced with #1840 (`# export ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER  #Optional, by default is true; can be set to false if you don't want to use --proxy-user option with Spark interpreter when impersonation enabled`), is that intended? If so, we will need to update the doc.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java [hunk 2]
```diff
@@ -196,10 +199,10 @@ private boolean isYarnMode() {
   private String toShellFormat(String value) {
     if (value.contains("\'") && value.contains("\"")) {
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java]: `"\'"` -> `"'"`

Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java]: fixed

File: bin/interpreter.sh [hunk 3]
```diff
@@ -207,14 +211,15 @@ fi
 
 if [[ -n "${SPARK_SUBMIT}" ]]; then
     if [[ -n "$ZEPPELIN_IMPERSONATE_USER" ]] && [[ "$ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER" != "false" ]];  then
-       INTERPRETER_RUN_COMMAND+=' '` echo ${SPARK_SUBMIT} --class ${ZEPPELIN_SERVER} --driver-class-path \"${ZEPPELIN_INTP_CLASSPATH_OVERRIDES}:${ZEPPELIN_INTP_CLASSPATH}\" --driver-java-options \"${JAVA_INTP_OPTS}\" ${SPARK_SUBMIT_OPTIONS} ${ZEPPELIN_SPARK_CONF} --proxy-user ${ZEPPELIN_IMPERSONATE_USER} ${SPARK_APP_JAR} ${CALLBACK_HOST} ${PORT} ${INTP_PORT}`
+       INTERPRETER_RUN_COMMAND+=' '` echo ${SPARK_SUBMIT} --class ${ZEPPELIN_SERVER} --driver-class-path \"${ZEPPELIN_INTP_CLASSPATH_OVERRIDES}:${ZEPPELIN_INTP_CLASSPATH}\" --driver-java-options \"${JAVA_INTP_OPTS}\" ${SPARK_SUBMIT_OPTIONS} ${ZEPPELIN_SPARK_CONF} ${SPARK_APP_JAR} ${CALLBACK_HOST} ${PORT} ${INTP_PORT}`
```
Comment (prabhjyotsingh) [inline bin/interpreter.sh]: Both if and else are same!

**Comment by zjffdu** [comment]
Thanks @prabhjyotsingh comments has been addressed. And CI passed

**Comment by prabhjyotsingh** [comment]
Rest, looks good.

**Comment by zjffdu** [comment]
Thanks, Merge if no more comment

**Pull request closed without merging**